### ISSUE TITLE
Infinite editing doc type throws JS error when template is allowed

### ISSUE THREAD

**Issue by hemraker** [umbraco/Umbraco-CMS#7181]
When editing a Document Type with Infinite Editing, a JS error is thrown in the console if a template is allowed on the Document Type.

## Umbraco Version
8.1.5+ (Tested 8.1.5, 8.2.1, 8.3)

### Specifics

Edit any document type, that has templates allowed, through infinite editing. Saving the doct type will produce following JS error:
```
Possibly unhandled rejection: Could not find the tree templates angular.js:15536:41
    Angular 8
        consoleLog
        get
        processChecks
        $digest
        $apply
        done
        completeRequest
        requestLoaded
```

### Steps to reproduce

1. Install 8.1.5+ with starterkit
2. Go to Home content node
3. Go to the Info tab and open the Home document type
4. Press Save and Close
5. Error is thrown in console

### Expected result

I expected the doctype to be saved without throwing an error.

### Actual result

Doc type is saved but with error in the JS console

---
_This item has been added to our backlog [AB#3877](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/3877)_

**Comment by kjac** [comment]
I can reproduce this... and I've found the error. I'll submit a PR tonight 👍

**Comment by kjac** [comment]
PR in #7183

**Comment by nielslyngsoe** [comment]
Though KJACs request follows the way this issue has been solved before in that file, I have chosen to grab the option to solve this in the right way to avoid having issues in the future.

So a new PR for this issue:
https://github.com/umbraco/Umbraco-CMS/pull/7207

**Comment by warrenbuckley** [comment]
Closed by PR #7207
